contrib/lib/src/ron.rs

```diff
@@ -0,0 +1,179 @@
+use std::ops::{Deref, DerefMut};
+use std::io;
+use std::iter::FromIterator;
+
+use tokio_io::AsyncReadExt;
+
+use rocket::request::Request;
+use rocket::outcome::Outcome::*;
+use rocket::data::{Transform::*, Transformed, Data, FromData, TransformFuture, FromDataFuture};
+use rocket::response::{self, Responder, content};
+use rocket::http::Status;
+
+use serde::{Serialize, Serializer};
+use serde::de::{Deserialize, Deserializer};
+
+
+#[derive(Debug)]
+pub struct Ron<T>(pub T);
+
+impl<T> Ron<T> {
+    #[inline(always)]
+    pub fn into_inner(self) -> T {
+        self.0
+    }
+}
+
+
+//// No idea what the default limit for ron is. Using Json's default of 1MB here.
+const LIMIT: u64 = 1 << 20;
+
+#[derive(Debug)]
+pub enum RonError<'a> {
+    Io(io::Error),
+
+    Parse(&'a str, ron::de::Error),
+}
+
+impl<'a, T: Deserialize<'a>> FromData<'a> for Ron<T> {
+    type Error = RonError<'a>;
+    type Owned = String;
+    type Borrowed= str;
+ 
+    fn transform(r: &Request<'_>, d: Data) -> TransformFuture<'a, Self::Owned, Self::Error> {
+        let size_limit = r.limits().get("ron").unwrap_or(LIMIT);
+        Box::pin(async move {
+            let mut s = String::with_capacity(512);
+            let mut reader = d.open().take(size_limit);
+            match reader.read_to_string(&mut s).await {
+                Ok(_) => Borrowed(Success(s)),
+                Err(e) => Borrowed(Failure((Status::BadRequest, RonError::Io(e))))
+            }
+        })
+    }
+
+    fn from_data(_: &Request<'_>, o: Transformed<'a, Self>) -> FromDataFuture<'a, Self, Self::Error> {
+        Box::pin(async move {
+            let string = try_outcome!(o.borrowed());
+            match ron::de::from_str(&string) {
+                Ok(v) => Success(Ron(v)),
+                Err(e) => {
+                    error_!("Couldn't parse RON body: {:?}", e);
+                    Failure((Status::BadRequest, RonError::Parse(string, e)))
+                }
+            }
+        })
+    }
+}
+
+
+impl<'r, T: Serialize> Responder<'r> for Ron<T> {
+    fn respond_to(self, req: &'r Request<'_>) -> response::ResultFuture<'r> {
+        match to_string(&self.0) {
+            Ok(string) => Box::pin(async move { Ok(content::Ron(string).respond_to(req).await.unwrap()) }),
+            Err(e) => Box::pin (async move {
+                error_!("RON failed to serialize: {:?}", e);
+                Err(Status::InternalServerError)
+            })
+        }
+    }
+}
+
+fn to_string<T>(value: &T) -> ron::ser::Result<String>
```

Comment:
`to_string` confused me earlier (I thought it referred to the function in `ron` until I got the). Can we give this a clearer name or inline it?